app/src/main/java/comcmput301f16t01/github/carrier/Users/NewVehicleActivity.java

```diff
@@ -0,0 +1,125 @@
+        EditText yearEditText = (EditText) findViewById(R.id.editText_vehicleYear);
+        EditText colorEditText = (EditText) findViewById(R.id.editText_vehicleColor);
+        EditText otherEditText = (EditText) findViewById(R.id.editText_vehicleOther);
+
+        //This is the funtion that will make a pretty string out of all the info given
+        String vehicleDescription = concatenateVehicleDescription(makeEditText.getText().toString(),
+                modelEditText.getText().toString(), yearEditText.getText().toString(),
+                colorEditText.getText().toString(), otherEditText.getText().toString());
+
+        //we will unpack for the bundle that was sent to us
+        //from the new user activity
+        Bundle bundle = getIntent().getExtras();
+        String username = bundle.getString("username");
+        String phone = bundle.getString("phone");
+        String email = bundle.getString("email");
+
+        //create a new user
+        //this is what adds to elastic search
+        UserController uc = new UserController();
+        uc.createNewUser(username, email, phone, vehicleDescription);
+
+        //go tot the main activity with this function
+        submitNewUser(username);
+    }
+
+    /**
+     * This function is what brings the user to the requests page
+     * It toasts that they have logged in successfully
+     * @param username The user we are trying to create
+     */
+    public void submitNewUser( String username) {
+        // Save username to file
+        LoginMemory lm = new LoginMemory( this );
+        lm.saveUsername( username );
+
+        // Log in to CARier
+        String welcome = "Welcome to CARrier, " + username + "!";
+        Intent intent = new Intent(this, MainActivity.class);
+        Toast.makeText(this, welcome, Toast.LENGTH_SHORT).show();
+        startActivity(intent);
+        this.finish();
+    }
+
+
+    /**
+     * This function will take these multiple strings and concatentate them into a nice string that is easy to store and display
+     *
+     * @param make      The make of the user's vehicle
+     * @param model     The model of the user's vehicle
+     * @param year      The year of the user's vehicle
+     * @param color     The color of the user's vehicle
+     * @param other     The notes the user added to their vehicle
+     * @return A string with all the info that is easy to display and read
+     */
+    private String concatenateVehicleDescription(String make, String model, String year,
+                                                 String color, String other) {
+        String vehicleDescription = "";
+
+        // If all squares are empty then just leave the string empty
+        if ( make.equals("") && model.equals("") && year.equals("")
+        && color.equals("") && other.equals("") ) {
+            vehicleDescription = "No description entered";
+        }
+        //else we need to piece the string together
+        else {
```

Comment:
I think actually (after looking on kijiji) that it is more common to have the year first (i.e. 2003 Kia Rio)